pkg/util/ring.go

```diff
@@ -9,3 +13,21 @@ func TokenFor(userID, labels string) uint32 {
 	_, _ = h.Write([]byte(labels))
 	return h.Sum32()
 }
+
+// IsInReplicationSet will query the provided ring for the provided key
+// and see if the provided address is in the resulting ReplicationSet
+func IsInReplicationSet(r *ring.Ring, ringKey uint32, address string) (bool, error) {
+	bufDescs, bufHosts, bufZones := ring.MakeBuffersForGet()
+	rs, err := r.Get(ringKey, ring.WriteNoExtend, bufDescs, bufHosts, bufZones)
```

Comment:
```suggestion
	rs, err := r.Get(ringKey, ring.Write, bufDescs, bufHosts, bufZones)
```
If the first 2 nodes for a token aren't active (i.e. during a rollout), `ring.WriteNoExtend` won't keep iterating the ring until an active node is picked, so I think we should use `ring.Write` here instead. This is the same idea from https://github.com/grafana/loki/pull/4607